goalie/optimisation.py

```diff
@@ -151,19 +169,73 @@ def minimise(self):
             self.progress["qoi"].append(J)
             self.progress["gradient"].append(dJ)
 
+            if self.adaptive:
```

Comment:
Since this whole section is effectively copied from `GoalOrientedMeshSeq`, we should introduce a new method there for it and call it here. It could be named `fixed_point_iteration_step`, for example. A similar thing could be done for the other `MeshSeq` implementations.